### ISSUE TITLE
Bug: Adding a repo presents no response

### ISSUE THREAD

**Issue by bdougie** [open-sauced/open-sauced#832]
### Describe the bug

I have noticed an issue and wanted to document them so they can get a fix. 

1. When adding a repo, the goals cache takes 30seconds to update, leaving the user to wonder if the repo they added was indeed added.  

### To Reproduce

1. Go to repo list dashboard
2. Click on add a repo
3. Scroll down to wait for it show

### Expected behavior

Adding a repo should take them to the repo page, so they begin interaction. 

### Screenshots

![test2](https://user-images.githubusercontent.com/5713670/94341371-9e6bfb00-ffbd-11ea-9cf5-9b9f89702494.gif)

**Comment by filiptronicek** [comment]
Any ideas on how it would be best to make the UX better? Mine was, that we could do a "loading" page, that will be a placeholder for the repo until it is ready

**Comment by bdougie** [comment]
The issue with a loading page is that the action takes 21 seconds to run. This may need something that checks the action in the background and lets the user know it is working.

example of the action run: https://github.com/bdougie/open-sauced-goals/actions/runs/292281621

**Comment by sgrove** [comment]
It might be best to use a persisted query for this and hopefully remove the need for `data.json` - I'm not 100% sure what data needs to be pulled out, but to just start with e.g. star count:

```graphql
query GitHubRepoStarGazerCountQuery(
  $repoOwner: String!
  $repoName: String!
) {
  gitHub {
    repository(name: $repoName, owner: $repoOwner) {
      stargazers(
        first: 10
        orderBy: { field: STARRED_AT, direction: DESC }
      ) {
        totalCount
      }
    }
  }
}
```

Here's me persisting the query (and caching the result per-repo for 12 hours so I don't hit any rate limits):
![Screen Shot 2020-10-06 at 3 16 24 PM](https://user-images.githubusercontent.com/35296/95265875-ef15fc00-07e6-11eb-9fec-e59a977012bd.png)

Then you can just query it on page load for a user:
```bash
curl -X POST https://serve.onegraph.com/graphql?app_id=63e9fc1a-ca32-40c9-8440-06540102e378 --data '{"doc_id": "aca405f1-4655-46f2-94c7-f039eb72e2ab", "variables": {"repoOwner": "sgrove", "repoName": "blog"} }'

> {"data":{"gitHub":{"repository":{"stargazers":{"totalCount":4}}}}}
```

**Comment by CodeItQuick** [comment]
Yeah this is a bit of a usability issue - I accidentally added two copies of the same repo and now I am nervous to remove it from the system (the ominous "Delete Repository" button makes me worry I'll self-destruct the repo). 

It takes quite a while to sync, and doesn't give feedback that it is "working". You can also add two of the same repository (this should be blocked with an error being fed back to the user).

Also as far as error reporting - When a 404 or 500 occurs it should spit back the error in the dashboard (like in a red box), as opposed to me looking through the console or network tab in the inspection wizard.

**Comment by mtfoley** [comment]
@bdougie the React stuff is still a bit out of my depth, but I think there's some issue with `goalsReducer` that deals with this. Adding `console.log` all up and down the chain in `src/components/RepositoryGoals.js`, I suspect there's an issue somewhere in all the callbacks and calls to [dispatch](https://github.com/open-sauced/open-sauced/blob/main/src/components/RepositoryGoals.js#L47), because this `updatedRepos` [function](https://github.com/open-sauced/open-sauced/blob/main/src/components/RepositoryGoals.js#L36) right before that doesn't get called, and I think that function generates what the reducer needs to see.

**Comment by mtfoley** [comment]
Related to #986, IMO (state management issues)
